drivers/gpio/gpio_shell.c

```diff
@@ -56,17 +58,33 @@ static int cmd_gpio_conf(const struct shell *sh, size_t argc, char **argv)
 			return 0;
 		}
 	} else {
-		shell_error(sh, "Wrong parameters for conf");
-		return -ENOTSUP;
+		shell_error(sh, "Wrong parameters for GPIO configuration");
+		return -EINVAL;
 	}
 
 	dev = device_get_binding(argv[args_indx.port]);
 
 	if (dev != NULL) {
-		index = (uint8_t)atoi(argv[args_indx.index]);
-		shell_print(sh, "Configuring %s pin %d",
+		cfg = (const struct gpio_driver_config *)dev->config;
+		if (cfg == NULL) {
+			shell_error(sh, "Device Configuration is NULL");
+			return -EIO;
+		}
+		if (index < find_msb_set(cfg->port_pin_mask)) {
+			shell_print(sh, "Configuring %s pin %u",
 			    argv[args_indx.port], index);
```

Comment:
Is this change necessary? It seems to be reaching into the implementation rather than using the interface. Shouldn't the new driver Just Work (TM) with `gpio_pin_configure()`

Reply:
same explanation for other comment